Framework/Source/API/D3D/D3D12/LowLevel/D3D12RootSignature.cpp

```diff
@@ -130,20 +130,9 @@ namespace Falcor
         for (size_t i = 0 ; i < mDesc.mSets.size() ; i++)
         {
             const auto& set = mDesc.mSets[i];
-            assert(set.getRangeCount() == 1);
-            uint32_t byteOffset;
-            if (set.getRangeCount() == 1 && set.getRange(0).type == DescType::Cbv)
-            {
-                convertCbvSet(set, rootParams[i]);
-                byteOffset = 8;
-            }
-            else
-            {
-                convertDescTable(mDesc.mSets[i], rootParams[i], d3dRanges[i]);
-                byteOffset = 4;
-            }
+            convertDescTable(mDesc.mSets[i], rootParams[i], d3dRanges[i]);
             mElementByteOffset[i] = mSizeInBytes;
-            mSizeInBytes += byteOffset;
+            mSizeInBytes += 4;
```

Comment:
Okay, so you are eliminating an "optimization" here that turned a descriptor set of a single CB into a root CB. Does that mean that there is another path I can use when I really *do* want a root CB?

Reply:
Not yet. There's an open issue to figure out how to expose the root-descriptors and root-constants. This change doesn't affect the performance